pallav/fashionTech/pallavBanka/parkingAssignment/Parking.java

```diff
@@ -1,2 +1,173 @@
-package com.fashionTech.pallavBanka.parkingAssignment;
+package fashionTech.pallavBanka.parkingAssignment;
 
+import java.util.Date;
+import java.util.Timer;
+
+abstract class parking{
+    void parkVehicle(vehicle v){
+        int regno;
+        short floor;
+        int slotid;
+        regno=v.regno;
+        floor=v.floor;
+        slotid=v.slotid;
+    }
+    void getVehicle(vehicle v,int regno){
+        v.regno=regno;
+        if(v.outTime != null) {
+            //difference in time in milliseconds
+            long diff = v.outTime.getTime() - v.inTime.getTime();
+            long diffHours = diff / (60 * 60 * 1000) % 24;
+            //charges Rs20/hour
+            float totalCharges = 20 * diffHours;
+            System.out.println("Your floor is:" + v.floor + "Your Slot ID is:" + v.slotid + "Your charges are:" + totalCharges);
+        }else
+            System.out.println("The vehicle hasn't left the premises.");
+    }
+}
+class vehicle extends parking{
```

Comment:
Vehicle class should not extend parking, vehicle and parking are totally unrelated objects.